src/Servers/HttpSys/test/FunctionalTests/Listener/Utilities.cs

```diff
@@ -114,8 +114,25 @@ internal static HttpSysListener CreateServerOnExistingQueue(AuthenticationScheme
         internal static async Task<RequestContext> AcceptAsync(this HttpSysListener server, TimeSpan timeout)
         {
             var factory = new TestRequestContextFactory(server);
-            var acceptContext = new AsyncAcceptContext(server, factory);
-            var acceptTask = server.AcceptAsync(acceptContext).AsTask();
+            using var acceptContext = new AsyncAcceptContext(server, factory);
+            
+            async Task<RequestContext> AcceptAsync()
+            {
+                while (true)
+                {
+                    var requestContext = await server.AcceptAsync(acceptContext);
+
+                    if (server.ValidateRequest(requestContext) && server.ValidateAuth(requestContext))
+                    {
+                        requestContext.InitializeFeatures();
+                        return requestContext;
+                    }
+
+                    continue;
```

Comment:
Can be removed, the loop will start over without this too 😉 
Or had you something different in mind?

Reply:
Oops